src/revwalk.c

```diff
@@ -449,6 +453,45 @@ static int limit_list(git_commit_list **out, git_revwalk *walk, git_commit_list
 	return 0;
 }
 
+static int get_one_revision(git_commit_list_node **out, git_revwalk *walk, git_commit_list **list)
+{
+	int error;
+	git_commit_list_node *commit;
+
+	while(true) {
+		commit = git_commit_list_pop(list);
+		if (!commit) {
+			giterr_clear();
+			return GIT_ITEROVER;
+		}
+
+		/*
+		 * If we did not run limit_list and we must add parents to the
+		 * list ourselves.
+		 */
+		if (!walk->limited) {
+			if ((error = add_parents_to_list(walk, commit, list)) < 0)
+				return error;
+		}
+
+		*out = commit;
+		return 0;
+	}
+}
+
+static int get_revision(git_commit_list_node **out, git_revwalk *walk, git_commit_list **list)
```

Comment:
There's not really a difference to `get_one_revision` right now, is there?

Reply:
Right now it just calls the lower-level function but extra features would get built here.

Reply:
I'd say we should avoid having two equivalent function calls for now, though. It can still easily be extended later when implementing said extra features